src/lib/computeRollbackQuery.ts

```diff
@@ -24,6 +24,14 @@ function computeRollbackSet(
 
   return Object.entries(flattenBackupDocument).reduce(
     (rollbackSet, [key, value]) => {
+      const parentKeyToFullyRestore = setPropertiesDuringUpdate.find(
+        (userSet) => key.startsWith(`${userSet}.`),
+      );
+      if (parentKeyToFullyRestore && parentKeyToFullyRestore in backup) {
```

Comment:
@hugop95 You mean `&& parentKeyToFullyRestore in backup` was the right fix for https://github.com/360Learning/mongo-bulk-data-migration/pull/15
Good usage of `in` here

Reply:
@pp0rtal I think it is. But I don't know if we cover all cases. If all checks runs successfully, I think it's safe for now. But I wonder if we are not missing another case.

Reply:
@hugop95 Thanks to enforced `doRollbackAndAssertForInitialState` it's safe for those who use it at least.
For anything too much core-releated (auto rollback) next time I'll do a `-rc` version to batch those kind of tricky updates.